internal/pkg/handler/updated-handler.go

```diff
@@ -20,11 +32,210 @@ func (r ResourceUpdatedHandler) Handle() error {
 		// process resource based on its type
 		if _, ok := r.Resource.(*v1.ConfigMap); ok {
 			logrus.Infof("Performing 'Updated' action for resource of type 'configmap'")
+			rollingUpgrade(r, "configmaps", "deployments")
+			rollingUpgrade(r, "configmaps", "daemonsets")
+			rollingUpgrade(r, "configmaps", "statefulSets")
 		} else if _, ok := r.Resource.(*v1.Secret); ok {
 			logrus.Infof("Performing 'Updated' action for resource of type 'secret'")
+			rollingUpgrade(r, "secrets", "deployments")
+			rollingUpgrade(r, "secrets", "daemonsets")
+			rollingUpgrade(r, "secrets", "statefulSets")
 		} else {
-			logrus.Warnf("Invalid resource: Resource should be 'Secret' or 'Configmap' but found %v", r.Resource)
+			logrus.Warnf("Invalid resource: Resource should be 'Secret' or 'Configmap' but found, %v", r.Resource)
+		}
+	}
+	return nil
+}
+
+func rollingUpgrade(r ResourceUpdatedHandler, resourceType string, rollingUpgradeType string) {
+	client, err := kube.GetClient()
+	if err != nil {
+		logrus.Fatalf("Unable to create Kubernetes client error = %v", err)
+	}
+	var namespace, name, sshData, envName string
+	if resourceType == "configmaps" {
+		namespace = r.Resource.(*v1.ConfigMap).Namespace
+		name = r.Resource.(*v1.ConfigMap).Name
+		sshData = helper.ConvertConfigmapToSHA(r.Resource.(*v1.ConfigMap))
+		envName = "_CONFIGMAP"
+	} else if resourceType == "secrets" {
+		namespace = r.Resource.(*v1.Secret).Namespace
+		name = r.Resource.(*v1.Secret).Name
+		sshData = helper.ConvertSecretToSHA(r.Resource.(*v1.Secret))
+		envName = "_SECRET"
+	}
+
+	if rollingUpgradeType == "deployments" {
+		rollingUpgradeForDeployment(client, namespace, name, sshData, envName)
```

Comment:
`rollingUpgradeForDeployment`, `rollingUpgradeForDaemonSets`, `rollingUpgradeForStatefulSets` are almost identical to each other. Remove redundancy and figure out a way to make it more generic than copy pasting methods and just changing the resource type

Reply:
Also think of a better name for these functions or just drop the `for` if you still want to use these functions. But I think they can be combined into 1 function with no duplication logic